src/app/components/layouts/tag/terra-tag.component.spec.ts

```diff
@@ -1,11 +1,252 @@
-// /* tslint:disable:no-unused-variable */
-//
-// import { TestBed, async } from '@angular/core/testing';
-// import { PlentyTagComponent } from './plenty-tag.component';
-//
-// describe('Component: TerraTagComponent', () => {
-//   it('should create an instance', () => {
-//     let component = new PlentyTagComponent();
-//     expect(component).toBeTruthy();
-//   });
-// });
+import {
+    async,
+    ComponentFixture,
+    TestBed
+} from '@angular/core/testing';
+
+import { TerraTagComponent } from '../../../..';
+import { LocalizationModule } from 'angular-l10n';
+import { l10nConfig } from '../../../translation/l10n.config';
+import { HttpClientModule } from '@angular/common/http';
+import {
+    tagOne,
+    tagTwo
+} from '../../../testing/mock-tags';
+import { DebugElement } from '@angular/core';
+import { By } from '@angular/platform-browser';
+import { TerraTagNameInterface } from './data/terra-tag-name.interface';
+
+describe('TerraTagComponent', () =>
+{
+    let component:TerraTagComponent;
+    let fixture:ComponentFixture<TerraTagComponent>;
+    let tagDiv:DebugElement;
+    const customClass:string = 'testClass';
+    const name:string = 'Test';
+
+    beforeEach(async(() =>
+    {
+        TestBed.configureTestingModule({
+            declarations: [
+                TerraTagComponent
+            ],
+            imports:      [
+                HttpClientModule,
+                LocalizationModule.forRoot(l10nConfig)
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() =>
+    {
+        fixture = TestBed.createComponent(TerraTagComponent);
+        component = fixture.componentInstance;
+
+        tagDiv = fixture.debugElement.query(By.css('div.tag'));
+    });
+
+    it('should create', () =>
+    {
+        expect(component).toBeTruthy();
+    });
+
+    it('should be initialized', () =>
+    {
+        expect(component.inputIsTaggable).toBe(false);
+        expect(component.inputCustomClass).toBeUndefined();
+        expect(component.inputColor).toBeUndefined();
+        expect(component.inputBadge).toBeUndefined();
+        expect(component.name).toBeUndefined();
+        expect(component.names).toBeUndefined();
+        expect(component.inputIsTagged).toBe(false);
+        expect(component.isClosable).toBe(false);
+        expect(component.tagId).toBeUndefined();
+    });
+
+    it('`inputColor` should be equal to background-color style', () =>
+    {
+        expect(tagDiv.styles['background-color']).toBeFalsy(); // no background color set
+
+        component.inputColor = tagOne.color;
+
+        fixture.detectChanges();
+
+        expect(tagDiv.styles['background-color']).toEqual(component.inputColor);
```

Comment:
zur Sicherheit hier auf Gleichheit mit tagOne.color prüfen:
```suggestion
        expect(tagDiv.styles['background-color']).toEqual(tagOne.color);
```